plugin/cds-hooks/src/main/java/org/opencds/cqf/ruler/cdshooks/dstu3/CdsHooksUtil.java

```diff
@@ -0,0 +1,110 @@
+package org.opencds.cqf.ruler.cdshooks.dstu3;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.JsonParser;
+import ca.uhn.fhir.parser.LenientErrorHandler;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.hl7.fhir.dstu3.model.Parameters;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.ParameterDefinition;
+import org.hl7.fhir.dstu3.model.Resource;
+import org.hl7.fhir.dstu3.model.PlanDefinition;
+import org.opencds.cqf.ruler.cdshooks.request.CdsHooksRequest;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.ArrayList;
+
+import static org.opencds.cqf.ruler.utility.dstu3.Parameters.newParameters;
+import static org.opencds.cqf.ruler.utility.dstu3.Parameters.newPart;
+
+public class CdsHooksUtil {
+
+    private static final JsonParser parser = new JsonParser(FhirContext.forDstu3(), new LenientErrorHandler());
```

Comment:
Used the cached version:
new JsonParser(FhirContext.forDstu3Cached(), new LenientErrorHandler());

Reply:
Also, the JsonParser should be created per-request. It's not thread-safe.